pkg/hvutils/hvutils_test.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hvutils
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_getInstanceTypeFromTags(t *testing.T) {
+	tests := []struct {
+		name string
+		tags []string
+		want string
+		err  error
+	}{
+		{
+			name: "empty slice returns empty string",
+			tags: []string{},
+			want: "",
+			err:  errors.New(errors.New("No instance-type tags found on deviceId=1").Error()),
+		},
+		{
+			name: "invalid label value will be skipped",
+			tags: []string{"instance-type=&"},
+			want: "",
+			err:  errors.New(errors.New("deviceID=1 has invalid tag \"&\" a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')").Error()),
+		},
+		{
+			name: "valid label value will be used",
+			tags: []string{"instance-type=abc"},
```

Comment:
Add maybe a new test that has multiple tags, just to be sure.

Reply:
There is already a test wiht multiple tags:

```
		{
			name: "two labels",
			tags: []string{"instance-type=abc", "instance-type=abc"},
			want: "",
			err:  errors.New(errors.New("More than one instance-type tags found on deviceId=1: [abc abc]").Error()),
		},
	}
```

Have you seen this?

Reply:
I mean multiple tags, also different than instance-type!

Reply:
Now I understood what you meant. I added some unrelated hv-labels in 6bf5c5f